packages/react/src/products/hooks/useProductGrouping.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  buildQueryStringFromObject,
+  fetchProductGrouping,
+  getProductGrouping,
+  getProductGroupingError,
+  isProductGroupingFetched,
+  isProductGroupingLoading,
+  ProductEntity,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useAction } from '../../helpers';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { UseProductGroupingOptions } from './types';
+
+const useProductGrouping = (
+  productId: ProductEntity['id'],
+  options: UseProductGroupingOptions = {},
+) => {
+  const {
+    fetchConfig,
+    enableAutoFetch = true,
+    fetchQuery = { pageIndex: 1 },
+  } = options;
+  const fetchAction = useAction(fetchProductGrouping);
+
+  const error = useSelector((state: StoreState) =>
+    getProductGroupingError(state, productId),
+  );
+  const isLoading = useSelector((state: StoreState) =>
+    isProductGroupingLoading(state, productId),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isProductGroupingFetched(state, productId),
+  );
+  const queryString = fetchQuery && buildQueryStringFromObject(fetchQuery);
+  const hash = queryString ? queryString : '!all';
+  const productGrouping = useSelector((state: StoreState) =>
+    getProductGrouping(state, productId, hash),
+  );
+
+  const shouldLoadProductGrouping =
+    enableAutoFetch && !isLoading && !error && !productGrouping;
```

Comment:
Actually you only need this:

```suggestion
    enableAutoFetch && !isLoading && !isFetched;
```
This is because `isFetched` will be true when there is an error or a result in the store, but it will be false when the it is loading so we need to explicitly check that it is not loading as well.

Reply:
I'll change it